circuit_knitting/cutting/cut_wire_to_move.py

```diff
@@ -0,0 +1,84 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Function to transform a :class:`.CutWire` instruction to a :class:`.Move` instruction."""
+from __future__ import annotations
+
+from itertools import groupby
+from qiskit.circuit import Qubit, QuantumCircuit
+from circuit_knitting.cutting.instructions.move import Move
+
+
+def transform_to_move(circuit: QuantumCircuit) -> QuantumCircuit:
```

Comment:
I realized late last week that it would be even better for this function to return a `Move` instruction that has already been wrapped in a `TwoQubitGate`.  So, instead of `new_inst = Move()`, it would be something like `new_inst = TwoQubitQPDGate(QPDBasis.from_gate(Move()), label="cut_move")`.  (We should probably make a shorter way for obtained a wrapped instruction like this eventually, after some discussion, but this is the method that currently works.)

This change would allow the output of this function to be sent directly to `partition_problem` once #350 is merged.

With this in mind, I wonder if something like `place_cut_wires` would be a better name for this function.  (`preprocess_cut_wires` is another potential name that I thought of earlier.)  @caleb-johnson, what do you think?)

Reply:
> With this in mind, I wonder if something like `place_cut_wires` would be a better name for this function. (`preprocess_cut_wires` is another potential name that I thought of earlier.) @caleb-johnson, what do you think?)

I think my earlier name might actually be better, since the `CutWire`s are not being placed, they're actually being _re_-placed.  I still find the phrase `preprocess_cut_wires` to be a little bit vague though.  And it's really _processing_ the cut wires -- I called it preprocess only because it's meant to happen early in the workflow, e.g. before `partition_problem`.

Reply:
> I realized late last week that it would be even better for this function to return a `Move` instruction that has already been wrapped in a `TwoQubitGate`.

One competing idea to be to merge this function in its current form, since it adds immediate value.  Then, in the next release (v0.4), we could introduce the new function that I suggest (both of these functions could be implemented by calling a common private function, which would contain the guts of the current PR), and we could deprecate this function if it is no longer expected to provide value.  This might be a more straightforward path.

Reply:
We say in the docs we play fast and loose with the terms "cut/partition" to avoid using the term "decompose" when describing the act of setting a circuit up for QPD. I think it's a fair vaguery ;)

Reply:
I renamed this function to `transform_cuts_to_moves` in 1200e646e186f2f68cbe955bba39ea1fc44ce0c5.  The additional (future) function, which will return `TwoQubitQPDGate`s, will need a different name anyway.  `cut_wires` might indeed make the most sense for that function; we can decide later.